plugins/anx-udev/fu-anx-udev-device.c

```diff
@@ -0,0 +1,475 @@
+
+	intfs = g_usb_device_get_interfaces (usb_device, error);
+	if (intfs == NULL)
+	{
+		g_debug ("anx_udev:fu_anx_udev_device_find_interface,intfs == NULL");
+		return FALSE;
+	}
+	self->vid = g_usb_device_get_vid (usb_device);
+	self->pid = g_usb_device_get_pid (usb_device);
+	self->rev = g_usb_device_get_release (usb_device);
+	g_debug ("USB: VID:%04X, PID:%04X, REV:%04X", self->vid, self->pid,
+					self->rev);
+	for (guint i = 0; i < intfs->len; i++) {
+		GUsbInterface *intf = g_ptr_array_index (intfs, i);
+		if (g_usb_interface_get_class (intf) == BILLBOARD_CALSS &&
+			g_usb_interface_get_subclass (intf) == BILLBOARD_SUBCLASS &&
+			g_usb_interface_get_protocol (intf) == BILLBOARD_PROTOCOL) {
+			//GUsbEndpoint *ep;
+			g_autoptr(GPtrArray) endpoints = NULL;
+
+			endpoints = g_usb_interface_get_endpoints (intf);
+			if (NULL == endpoints)
+					continue;
+			/*if (endpoints->len == 0)
+			{
+					ep = endpoints;
+			}
+			else
+			{
+					ep = g_ptr_array_index (endpoints, 0);
+			}
+			g_debug ("anx_udev:fu_anx_udev_device_find_interface, NULL != endpoints");
+			self->iface_idx = g_usb_interface_get_number (intf);
+			self->ep_num = g_usb_endpoint_get_address (ep) & 0x7f;
+			self->chunk_len = g_usb_endpoint_get_maximum_packet_size (ep);
+            */
+			return TRUE;
+		}
+	}
+	g_set_error_literal (error,
+						FWUPD_ERROR,
+						FWUPD_ERROR_NOT_FOUND,
+						"no update interface found");
+	return FALSE;
+}
+
+static gboolean
+fu_anx_udev_device_probe (FuUsbDevice *device, GError **error)
+{
+	g_debug ("anx_udev:fu_anx_udev_device_probe");
+	if (!fu_anx_udev_device_find_interface (device, error)) {
+		g_prefix_error (error, "failed to find update interface: ");
+		return FALSE;
+	}
+	set_custom_version (0);
+	set_fw_version (0);
+	/* set name and vendor */
+	fu_device_set_summary (FU_DEVICE (device),
+							"Analogix Phoenix-Lite");
+	fu_device_set_vendor (FU_DEVICE (device), "Analogix Semiconductor Inc.");
+	/* success */
+	return TRUE;
+}
+
+
+static FuFirmware *
+fu_anx_udev_device_prepare_firmware (FuDevice *device,
+                                   GBytes *fw,
+                                   FwupdInstallFlags flags,
+                                   GError **error)
+{
+	g_autoptr(FuFirmware) firmware = fu_anx_udev_firmware_new ();
+	g_debug ("anx_udev:fu_anx_udev_device_prepare_firmware, flag = %d",
+							(gint)flags);
+	if (!fu_firmware_parse (firmware, fw, flags, error))
+		return NULL;
+	return g_steal_pointer (&firmware);
+}
+
+static gboolean program_flash (guint32 start_addr, guint32 total_len,
+								guint32 len, guint16 req_val,
+								FuDevice *device, guint32 base,
+								const guint8 *source_buf)
+{
+	static guint32 wrote_len = 0;
+	guint32 packet_count;
+	guint32 left_count;
+	guint32 packet_index = 0;
+	FuAnxUdevDevice *self = FU_ANX_UDEV_DEVICE (device);
+	packet_count = len/BILLBOARD_MAX_PACKET_SIZE;
+	left_count = len%BILLBOARD_MAX_PACKET_SIZE;
+	if (source_buf == NULL)
+			return FALSE;

```

Comment:
whitespace is way off here.
also this and many other lines in this function are causing CI problems.
You can run `./contrib/ci/check-null-false-returns.py` locally to reproduce them.